hs_swat_modelinstance/tests/test_swat_modelinstance_metadata.py

```diff
@@ -0,0 +1,824 @@
+        self.assertEquals(self.resSWATModelInstance.metadata.coverages.all().count(), 0)
+
+        # there shouldn't any format element
+        self.assertEquals(self.resSWATModelInstance.metadata.formats.all().count(), 0)
+
+        # there shouldn't any subject element
+        self.assertEquals(self.resSWATModelInstance.metadata.subjects.all().count(), 0)
+
+        # there shouldn't any contributor element
+        self.assertEquals(self.resSWATModelInstance.metadata.contributors.all().count(), 0)
+
+        # check that there are no extended metadata elements at this point
+        self.assertEquals(self.resSWATModelInstance.metadata.model_output, None)
+        self.assertEquals(self.resSWATModelInstance.metadata.executed_by, None)
+        self.assertEquals(self.resSWATModelInstance.metadata.model_input, None)
+        self.assertEquals(self.resSWATModelInstance.metadata.model_parameter, None)
+        self.assertEquals(self.resSWATModelInstance.metadata.model_method, None)
+        self.assertEquals(self.resSWATModelInstance.metadata.simulation_type, None)
+        self.assertEquals(self.resSWATModelInstance.metadata.model_objective, None)
+
+        # create #######################################################################################################
+        # create model_output ####
+        self.resSWATModelInstance.metadata.create_element('ModelOutput', includes_output=False)
+        modeloutput_element = self.resSWATModelInstance.metadata.model_output
+        self.assertEquals(modeloutput_element.includes_output, False)
+        # multiple ModelOutput elements are not allowed - should raise exception
+        with self.assertRaises(IntegrityError):
+            self.resSWATModelInstance.metadata.create_element('ModelOutput', includes_output=True)
+        with self.assertRaises(IntegrityError):
+            self.resSWATModelInstance.metadata.create_element('ModelOutput', includes_output=False)
+        self.resSWATModelInstance.metadata.delete_element('ModelOutput',
+                                                          self.resSWATModelInstance.metadata.model_output.id)
+        self.assertEqual(self.resSWATModelInstance.metadata.model_output, None)
+        self.resSWATModelInstance.metadata.create_element('ModelOutput', includes_output=True)
+        modeloutput_element = self.resSWATModelInstance.metadata.model_output
+        self.assertEquals(modeloutput_element.includes_output, True)
+        # multiple ModelOutput elements are not allowed - should raise exception
+        with self.assertRaises(IntegrityError):
+            self.resSWATModelInstance.metadata.create_element('ModelOutput', includes_output=True)
+            self.resSWATModelInstance.metadata.create_element('ModelOutput', includes_output=False)
+
+        # create ExecutedBy ####
+        self.resSWATModelInstance.metadata.create_element('ExecutedBy', model_name=self.resGenModelProgram.short_id)

```

Comment:
From this test code it seems the implementation of the class ```ExecutedBy``` is probably confusing. Here you are creating an object of type ```ExecutedBy``` by passing the id of the resource for the attribute ```model_name``` from which one would assume that the ```model_name``` attribute is being used to store the resource id. But from line 174 it seems it actually stores the title of the resource and not resource id.